### PR TITLE
[NO TICKET] stylelint cleanup

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

This is a combination of fixing and ignoring the majority of CSS stylelint errors we've had. I've added PR comments to explain decisions.

I should note I left errors in packages/docs/src/styles/index.scss unaddressed as it might warrant taking a deeper look at it. Errors in this file are:

- `scss/at-import-no-partial-leading-underscore`
- `no-invalid-position-at-import-rule`

## How to test

1. Pull down branch
2. Run `yarn lint:stylelint`

You should only see errors relating to the index.scss file mentioned above.

### If this is a change to code:

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[ticket] title`
- [x] Ran `yarn type-check && yarn lint && yarn test` with no errors
- [x] Updated unit test snapshots and loki reference images **if necessary** with `yarn loki` && `yarn update-snapshots`

### If this is a change to documentation:

- [x] Checked for spelling and grammatical errors

### CODE REVIEW

File: packages/design-system/src/styles/components/_Alert.scss [hunk 1]
```diff
@@ -50,8 +50,8 @@
 .ds-c-alert__a11y-label {
   margin-right: 0.5rem;
 
-  // !important needed to overwrite screen reader utility styles
   @media (-ms-high-contrast: active), (forced-colors: active) {
+    /* stylelint-disable declaration-no-important -- needed to overwrite screen reader utility styles */
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_Alert.scss:54]: Inlined the comment as part of the [lint exception description](https://stylelint.io/user-guide/usage/options/#reportdescriptionlessdisables).

For a11y reasons, we have a visually hidden screen reader label to let users on AT know what kind of alert they're reading (since our alert icons aren't visible). HCM users also need a visual queue, so we overwrite the CSS rules when HCM is enabled. This cannot be refactored in a way to remove `!important` as that specificity is needed for overwriting the SR utility class.

File: packages/design-system/src/styles/components/_Button.scss [hunk 2]
```diff
@@ -83,7 +83,7 @@
 
     // Disable
     --backgroundColor--d: #{$button-alt__background-color--disabled};
-    --borderColor: #{$button-alt__border-color--disabled};
+    --borderColor--d: #{$button-alt__border-color--disabled};
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_Button.scss:86]: Discovered a hidden bug on Buttons

File: packages/design-system/src/styles/components/_FilterChip.scss [hunk 3]
```diff
@@ -81,12 +81,6 @@ $filter-chip-icon-size: 10px;
   .ds-c-icon--close-thin {
     height: $filter-chip-icon-size + 2;
     width: $filter-chip-icon-size + 2;
-
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_FilterChip.scss:84]: Found this style was unnecessary as `.ds-c-icon--close` has the same styles applied and the thin icon also received those styles as well. 

tldr; redundant styling.

File: packages/design-system/src/styles/components/_SingleInputDateField.scss [hunk 4]
```diff
@@ -8,6 +8,7 @@
 // the source rules.
 // @import 'react-day-picker/dist/style.css';
 
+/* stylelint-disable declaration-no-important, no-duplicate-selectors -- Ignore inlined 3rd party styles */
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_SingleInputDateField.scss:11]: As explained in the comment above, we're inlining 3rd party CSS to guarantee it loads in the correct order. Ignoring a few linter rules for the 3rd party styles only.

File: packages/design-system/src/styles/components/_SingleInputDateField.scss [hunk 5]
```diff
@@ -34,10 +35,8 @@
 /* Hide elements for devices that are not screen readers */
 .rdp-vhidden {
   box-sizing: border-box;
-  padding: 0;
   margin: 0;
   background: transparent;
-  border: 0;
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_SingleInputDateField.scss:40]: Removed redundant CSS as it was an easy thing to do.

File: packages/design-system/src/styles/components/_SvgIcon.scss [hunk 6]
```diff
@@ -62,7 +62,7 @@
   fill: $color-white;
 
   @media (-ms-high-contrast: active), (forced-colors: active) {
-    fill: Window !important;
+    fill: Window;
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_SvgIcon.scss:65]: Tested this locally and the `!important` didn't seem necessary

File: packages/design-system/src/styles/components/_Tabs.scss [hunk 7]
```diff
@@ -16,7 +16,7 @@
   padding: $spacer-3;
 
   &[aria-hidden='true'] {
-    display: none !important;
+    display: none;
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_Tabs.scss:19]: Tested locally and the `!important` didn't seem necessary.

TODO: Perhaps Tabs should be refactored to use `hidden` instead of `aria-hidden`? The `hidden` attr has the same effect as `aria-hidden` in addition to visually hiding the content when applied - which is what's being done with this CSS.

File: packages/design-system/src/styles/components/_TooltipIcon.scss [hunk 8]
```diff
@@ -20,8 +20,7 @@
   fill: $tooltip-icon__color;
   height: $tooltip-icon-size;
   left: 0;
-  // fixes clipping issue in Firefox, overrides vendor style
-  overflow: visible !important;
+  overflow: visible !important; /* stylelint-disable-line declaration-no-important -- fixes clipping issue in Firefox, overrides vendor style */
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_TooltipIcon.scss:23]: This CSS had a very specific comment explaining why the `!important` was needed, but I couldn't find an independent source saying this was still an issue with Firefox (found some indicator it was true with Safari however).

Local testing removing this rule didn't show any obvious breaking issues and I think it _could_ be removed, but want to revisit later with an official ticket to do better browser testing with a refactor.

File: packages/docs/src/styles/components/embeddedExamples.scss [hunk 9]
```diff
@@ -7,7 +7,7 @@
 }
 
 .example-list {
-  & article {
+  article {
```
Comment (zarahzachz) [inline packages/docs/src/styles/components/embeddedExamples.scss:10]: This use of the & isn't doing anything because the styles are nested.

File: packages/ds-medicare-gov/src/styles/components/_Navbar.scss [hunk 10]
```diff
@@ -1,3 +1,4 @@
+/* stylelint-disable declaration-no-important -- Temporarily ignored until refactor can happen */
```
Comment (zarahzachz) [inline packages/ds-medicare-gov/src/styles/components/_Navbar.scss:1]: Ignoring as it's `.m-c-` namespaced styles as they were added by outside teams and it's unknown what they're overwriting without doing a deeper dive.

File: packages/ds-medicare-gov/src/styles/components/_NavigationMenu.scss [hunk 11]
```diff
@@ -1,3 +1,4 @@
+/* stylelint-disable declaration-no-important -- Temporarily ignore until mgov overrides can be refactored */
```
Comment (zarahzachz) [inline packages/ds-medicare-gov/src/styles/components/_NavigationMenu.scss:1]: Ignoring as it's `.m-c-` namespaced styles as they were added by outside teams and it's unknown what they're overwriting without doing a deeper dive.

File: packages/ds-medicare-gov/src/styles/components/_SimpleFooter.scss [hunk 12]
```diff
@@ -1,3 +1,4 @@
+/* stylelint-disable declaration-no-important -- Temporarily ignore until mgov can be refactored */
```
Comment (zarahzachz) [inline packages/ds-medicare-gov/src/styles/components/_SimpleFooter.scss:1]: Ignoring as it's `.m-c-` namespaced styles as they were added by outside teams and it's unknown what they're overwriting without doing a deeper dive.

File: packages/ds-medicare-gov/src/styles/components/_override.dialog.scss [hunk 13]
```diff
@@ -1,3 +1,4 @@
+/* stylelint-disable declaration-no-important -- temporarily ignore until mgov can be refactored */
```
Comment (zarahzachz) [inline packages/ds-medicare-gov/src/styles/components/_override.dialog.scss:1]: It's unknown what mgov is overwriting and why - should be migrated up the chain for tokens? Need a ticket for deeper investigation

File: packages/ds-medicare-gov/src/styles/settings/variables/_layout.scss [hunk 14]
```diff
@@ -1 +1,3 @@
+/* stylelint-disable declaration-no-important -- Temporarily ignore until mgov can be refactored */
 $card-border-radius: 6px !important;
```
Comment (zarahzachz) [inline packages/ds-medicare-gov/src/styles/settings/variables/_layout.scss:2]: Maybe this needed to be migrated to tokens? Needs ticket to do deeper investigation.

File: packages/design-system/src/styles/components/_Button.scss (hunk 2 above)
Comment (pwolfert) [reply packages/design-system/src/styles/components/_Button.scss:86]: Nice!

File: packages/design-system/src/styles/components/_Tabs.scss (hunk 7 above)
Comment (pwolfert) [reply packages/design-system/src/styles/components/_Tabs.scss:19]: Maybe it would be worth adding a "TODO" comment in the code 😆

**Comment by zarahzachz** [comment]
I ran Loki off `master` and updated incorrect refs. Used `git switch` to move updated refs from `master` to this branch, and ran `yarn loki test`. No errors. 

Committed updated ref images as part of this PR.
